### PR TITLE
Make "hermit search" available outside an active env.

### PR DESCRIPTION
**Pull request by alecthomas**

### CODE REVIEW

File: app/commands.go [hunk 1]
```diff
@@ -615,15 +614,33 @@ type searchCmd struct {
...
+	var (
+		pkgs manifest.Packages
+		err  error
+	)
+	if env != nil {
+		err = env.Sync(l, false)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		pkgs, err = env.Search(l, s.Constraint)
```
Comment (quad) [inline app/commands.go:627]: What's the difference between `env.Search` and `state.Search`?

**Review by quad — APPROVED** [review]

File: app/commands.go (hunk 1 above)
Comment (alecthomas) [reply app/commands.go:627]: env.Search() will search sources from the environment config.

Comment (quad) [reply app/commands.go:627]:
```go
type Searchable {
  Sync(*ui.UI, bool) error
  Search(*ui.UI, Constraint) Packages[], error
}
```

Or something. I can't write Go in a textbox yet.

Comment (alecthomas) [reply app/commands.go:627]: That did cross my mind but I decided YAGNI.

Comment (alecthomas) [reply app/commands.go:627]: In general I usually add abstractions if there are greater than 2-3 usages.

**Pull request merged**